spec/takeaway_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'takeaway'
+require 'menu'
+
+describe Takeaway do
+  
+  let(:menu) { double :menu, item: "sausage", price: 4 }
+  subject(:takeaway) { described_class.new(menu) }
+  items = { sausage: 6, pie: 2 }
+  let(:textsender) { double :textsender }
+
+  context 'order'
+
+  describe '#view_menu' do  
+    it 'should display the menu' do
+      expect(subject.view_menu).to eq(:menu.display)
+    end  
+  end
+  
+  describe '#select_dish' do
+    it 'should add a dish to the basket' do
+      subject.select_dish(:pie, 2)
+      expect(subject.basket).to include({ pie: 2 })
+    end
+  end
+
+  describe '#total' do
+    #how can I assign { "sausage" => 4 } to the menu double?
```

Comment:
you could do `allow(menu).to receive(:menu).and_return({ "sausage" => 4 })` since the test is complaining that your total method is calling `#menu` on the menu double.